nmpc_mhe/aux/utils.py

```diff
@@ -5,11 +5,17 @@
 from pyomo.environ import *
 # from pyomo.dae import ContinuousSet, DerivativeVar
 from pyomo.core.base import Suffix, ConcreteModel, Var, Suffix, Constraint, ConstraintList, TransformationFactory
+from pyomo.core.base.set import BoundsInitializer
 from pyomo.opt import ProblemFormat
 from pyomo.core.base import numvalue
 from os import getcwd, remove
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.sparse import lil_matrix
+from scipy.sparse.linalg import splu
```

Comment:
We need to add scipy to the requirements.
Perhaps we also want to use `conda env create -f environment.yml`

Reply:
Yes, except for `scipy`, I also used `control` and `copy`. I think these should be the requirements as well. 
Is that create a virtual environment in conda?

Reply:
I am asking for `conda env create -f environment.yml`.

Reply:
Yes